### PR TITLE
Typeclass instances

### PR DESCRIPTION
**Pull request by jmchapman**
Status:

Supports very simple instances. More work needed to support anything beyond something of the form `C a` or possibly `C a b`.

### CODE REVIEW

File: Main.hs
Comment (jmchapman) [inline Main.hs:373]: I removed this as it doesn't seem to be used. The next arg Hs.Name () is just a rendering of the `QName` which is used instead.

**Review by UlfNorell — APPROVED** [review]
Nice!

**Comment by jmchapman** [comment]
If this is already useful to the work on the prelude then please merge it and I will build on it in another PR. Otherwise I can carry on here.

I thought I was at some point but perhaps something has changed in connection with protected branches and CI.

**Comment by UlfNorell** [comment]
I'll merge it and see if I can figure out the permissions...

Not sure what the issue is.. it says all 43 members (of the Agda org) has write access. I tried making a team specifically for this. Hopefully that fixes things.

**Comment by jmchapman** [comment]
> I'll merge it and see if I can figure out the permissions...

Thanks. that seems to have fixed it. I noticed I had permission to merge https://github.com/agda/agda2hs/pull/32.

**Pull request merged**
